core/src/main/java/io/kestra/core/utils/MathUtils.java

```diff
@@ -0,0 +1,37 @@
+package io.kestra.core.utils;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+public class MathUtils {
+    /**
+     * Rounds a double value to a specified number of decimal places.
+     *
+     * @param value The double value to be rounded.
+     * @param decimalPlaces The number of decimal places to round to.
+     * Must be a non-negative integer.
+     * @return The rounded double value.
+     * @throws IllegalArgumentException If decimalPlaces is negative.
+     */
+    public static double roundDouble(double value, int decimalPlaces) {
+        // Validate the input for decimalPlaces
+        if (decimalPlaces < 0) {
+            throw new IllegalArgumentException("The number of decimal places must be non-negative.");
+        }
+
+        // Convert the double to a BigDecimal using its String representation.
+        // This is crucial to avoid floating-point precision issues that can occur
+        // when a double is directly converted to BigDecimal or when performing
+        // arithmetic operations on doubles.
+        BigDecimal bd = new BigDecimal(String.valueOf(value));
```

Comment:
Beeing exact on statistic is not needed.
It's not an issue to do so, if it's not inside a hot path as it'll cost more...